### PR TITLE
Add package wireshark and dependencies

### PR DESCRIPTION
**Pull request by healther**
This is mostly a copy of the respective homebrew receipts, I don't have first hand knowledge of the options and/or dependencies. If someone knows/finds the version information I'd be happy to add them in.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libmaxminddb/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    version('1.3.2', '67a861965b30d045bf29a2126bcc05ed')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    def autoreconf(self, spec, prefix):
+        # FIXME: Modify the autoreconf method as necessary
+        autoreconf('--install', '--verbose', '--force')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmaxminddb/package.py]: If you didn't need to make any modifications to this I think the default in `AutotoolsPackage` should work, so you can remove this function.

**Comment by citibeth** [comment]
Does `wireshark` require root?  How does Spack deal with stuff like that?

If we have a good way, then we can also consider NetHack for Spack.  Because no software repo is complete without it...

**Comment by healther** [comment]
Don't ask me how it works in the background, but it doesn't require root for building or installation

File: var/spack/repos/builtin/packages/wireshark/package.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            args.append('-DBUILD_NGHTTP2=OFF')
+
+        return args
+
+    @run_after('install')
+    def symlink(self):
+        if self.spec.satisfies('platform=darwin'):
+            link(join_path(self.prefix,
+                           'Wireshark.app/Contents/MacOS/Wireshark'),
+                 join_path(self.prefix.bin, 'wireshark'))
```
[the remark this answers is not in the document]
Comment (healther) [reply var/spack/repos/builtin/packages/wireshark/package.py]: how far does this work? Can I reach every `folder/file` combination by `prefix.folder.file`?

File: var/spack/repos/builtin/packages/wireshark/package.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    variant('smi', default=False, description='Build with libsmi')
+    variant('libssh', default=False, description='Build with libssh')
+    variant('nghttp2', default=False, description='Build with nghttp2')
+    variant('qt', default=False, description='Build with qt')
+    variant('gtk3', default=False, description='Build with gtk3')
+    variant('gtk', default=False, description='Build with gtk')
+    variant('headers', default=True, description='Install headers')
+
+    depends_on('cares')
+    depends_on('doxygen')
```
[the remark this answers is not in the document]
Comment (healther) [reply var/spack/repos/builtin/packages/wireshark/package.py]: It builds it by default, I could investigate how to deactivate it, but I think it's better to keep the documentation around

File: var/spack/repos/builtin/packages/wireshark/package.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/wireshark/package.py]: Yes, as long as those folders/files are valid Python variable names. For example, it won't work if your filename contains periods or dashes, and it won't work if your filename starts with a number. See https://github.com/spack/spack/blob/develop/lib/spack/spack/util/prefix.py#L57 for how I implemented this.

**Pull request merged**